chainermn/iterators/multi_node_iterator.py

```diff
@@ -29,28 +41,38 @@ def __init__(self, actual_iterator, communicator, rank_master):
     def __next__(self):
         try:
             batch = self.actual_iterator.__next__()
-            stop = False
+            first_elem = batch[0]
+            valid_data_type = _is_valid_type(first_elem)
             is_paired_dataset = isinstance(batch, list) \
-                and isinstance(batch[0], tuple) and len(batch[0]) == 2
+                and isinstance(first_elem, tuple) and len(first_elem) == 2
+            stop = False
         except StopIteration:
             stop = True
+            valid_data_type = False
             is_paired_dataset = False
 
         is_new_epoch = self.actual_iterator.is_new_epoch
 
         # Notify the followings to slave iterators:
         # 1. whether stop signal is received before broadcasting data.
-        # 2. whether dataset is paired.
-        # 3. is_new_epoch.
-        # 4. current_position.
-        _info = numpy.ones((4, )) \
-            * [int(stop), int(is_paired_dataset),
+        # 2. whether type of batch element is valid.
+        # 3. whether dataset is paired.
+        # 4. is_new_epoch.
+        # 5. current_position.
+        _info = numpy.ones((5, )) \
+            * [int(stop), int(valid_data_type), int(is_paired_dataset),
                int(is_new_epoch),
                int(self.actual_iterator.current_position)]
         _info = _info.astype(numpy.float32)
         self.communicator.bcast(_info, root=self.rank_master)
 
-        if not stop:
+        if stop:
+            raise StopIteration
+        elif not valid_data_type:
+            raise RuntimeError('Multi node iterator supports numpy.float32 '
```

Comment:
[TypeError](https://docs.python.org/3/library/exceptions.html) should be raised here.

> Passing arguments of the wrong type (e.g. passing a list when an int is expected) should result in a TypeError,